Source/DafnyCore/Backends/Dafny/DafnyCodeGenerator.cs

```diff
@@ -31,16 +36,34 @@ class DafnyCodeGenerator : SinglePassCodeGenerator {
       return res;
     }
 
-    public DafnyCodeGenerator(DafnyOptions options, ErrorReporter reporter) : base(options, reporter) {
+    public DafnyCodeGenerator(DafnyOptions options, ErrorReporter reporter, bool preventShadowing = true) : base(options, reporter) {
       options.SystemModuleTranslationMode = CommonOptionBag.SystemModuleMode.Include;
       if (Options?.CoverageLegendFile != null) {
         Imports.Add("DafnyProfiling");
       }
+
+      testing = Environment.GetEnvironmentVariable("TEST_DAFNY") == "true";
```

Comment:
Is `TEST_DAFNY` meant to mean "output partial code on errors"? I was thinking we could add a proper option for that instead, since we'll want smithy-dafny to use that too, and it isn't really "testing Dafny"

Reply:
Indeed. I removed TEST_DAFNY and instead replaced it with the option `--emit-uncompilable-code`, let me know if that name is ok for you!